lib/hsa/mcwamp_hsa.cpp

```diff
@@ -1021,7 +1008,7 @@ class HSAQueue final : public KalmarQueue
     int getPendingAsyncOps() override {
         int count = 0;
         for (int i = 0; i < asyncOps.size(); ++i) {
-            auto asyncOp = asyncOps[i];
+            auto asyncOp = asyncOps[i]; // TODO _ shared_ptr copy, can we optimize?
```

Comment:
Actually, since you don't use the index, you could just write:

```cpp
for(auto&& asyncOp:asyncOps) {
        if (asyncOp != nullptr) {
            hsa_signal_t signal = *(static_cast <hsa_signal_t*> (asyncOp->getNativeHandle()));
            hsa_signal_value_t v = hsa_signal_load_relaxed(signal);
            if (v != 0) {
                ++count;
            }
        }
    }
}
```